ios/PolyPodApp/Core/UpdateNotification.swift

```diff
@@ -107,74 +89,67 @@ class UpdateNotification {
     }
     
     private static let notificationData = Data.load()
-        
-    private static func loadLastState(_ id: Int) -> State {
-        if id == 0 {
-            return .ALL_SEEN
-        }
-        
-        guard let (lastId, lastState) = LastUpdateNotification.read() else {
-            return .NOT_SEEN
-        }
-        
-        if id < lastId {
-            return .ALL_SEEN
-        }
-        
-        if id != lastId {
-            return .NOT_SEEN
-        }
-        
-        return State.parse(lastState) ?? .ALL_SEEN
-    }
     
-    let id = UpdateNotificationMockId.read() ?? notificationData.id
     let pushDelay = notificationData.pushDelay
     let title = notificationData.localizedTitle
     let text = notificationData.localizedText
     
-    private var cachedState: State
-    private var state: State {
-        get { cachedState }
-        set {
-            cachedState = newValue
-            LastUpdateNotification.write(id: id, state: cachedState.rawValue)
-        }
-    }
-    
+    private let raw: OpaquePointer
+
     init() {
-        cachedState = UpdateNotification.loadLastState(id)
+        let storage = UpdateNotificationStorage(
+            context: nil,
+            read_id: { _ in
+                UInt32(UpdateNotificationMockId.read() ?? UpdateNotification.notificationData.id)
+            },
+            read_last_id: { _ in
+                if let id = LastUpdateNotification.readId() {
+                    return Option32(tag: .init(rawValue: 0), .init(.init(some32: UInt32(id))))
+                } else {
+                    return Option32(tag: .init(rawValue: 1), .init(.init()))
+                }
+            },
+            read_last_state: { _ in
+                if let state = LastUpdateNotification.readLastState() {
+                    return OptionSeen(tag: OptionSeen_Tag.init(rawValue: 0),
+                                      .init(.init(some_seen: .init(rawValue: UInt32(state)))))
+                } else {
+                    return OptionSeen(tag: OptionSeen_Tag.init(rawValue: 1),
+                                      .init(.init()))
+                }
+            },
+            write_last: { _, id, state in
+                LastUpdateNotification.write(id: Int(id), state: Int(state.rawValue))
+            }
+        )
+        raw = new_update_notification(storage)
     }
     
     var showPush: Bool {
         get {
-            return state == .NOT_SEEN
+            return show_push(raw)
```

Comment:
To execute any logic, the api functions are called by passing the earlier created pointer.